### PR TITLE
add a few depenedent destroys for user..

### PR DESCRIPTION
**Pull request by robguthrie**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -96,10 +97,11 @@ class User < ActiveRecord::Base
 
   has_many :notifications, dependent: :destroy
   has_many :comments, dependent: :destroy
-  has_many :attachments
+  has_many :attachments, dependent: :destroy
```
Comment (s01us) [inline app/models/user.rb]: this could wreck documents and seems strange considering you're leaving comments?

Comment (robguthrie) [inline app/models/user.rb]: I'm not leaving comments. Deactivate is where we keep user data.. delete is where we do our best to remove everything and leave no dangling references. It's a work in progress, we do not do any user deletion right now

File: app/models/user.rb [hunk 2]
```diff
@@ -96,10 +97,11 @@ class User < ActiveRecord::Base
...
 
   has_one :deactivation_response,
-          class_name: 'UserDeactivationResponse'
+          class_name: 'UserDeactivationResponse',
+          dependent: :destory
```
Comment (gdpelican) [inline app/models/user.rb]: Typo!

Comment (robguthrie) [inline app/models/user.rb]: My all time most common rails typo! Thanks

**Pull request merged**